lib/gen_socket_client.ex

```diff
@@ -182,7 +182,7 @@ defmodule Phoenix.Channels.GenSocketClient do
           {:ok, encoded} ->
             transport.transport_mod.push(transport.transport_pid, encoded)
             {:ok, ref}
-          error -> error
+          _error -> {:error, :encoding_error}
```

Comment:
you can do that and keep the original error by using `{:error, {:encoding_error, error}}`